### PR TITLE
Porting Layoutlmv2 to Huggingface

### PR DESCRIPTION
**Pull request by monuminu**
# What does this PR do?

Fixes # (issue)
https://github.com/huggingface/transformers/issues/11932

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section? Yes
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.  https://github.com/microsoft/unilm/issues/325
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation). No
- [ ] Did you write any new necessary tests? No

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@inproceedings{Xu2020LayoutLMv2MP,
  title     = {LayoutLMv2: Multi-modal Pre-training for Visually-Rich Document Understanding},
  author    = {Yang Xu and Yiheng Xu and Tengchao Lv and Lei Cui and Furu Wei and Guoxin Wang and Yijuan Lu and Dinei Florencio and Cha Zhang and Wanxiang Che and Min Zhang and Lidong Zhou},
  booktitle = {Proceedings of the 59th Annual Meeting of the Association for Computational Linguistics (ACL) 2021},
  year      = {2021},
  month     = {August},
}

### CODE REVIEW

File: src/transformers/models/layoutlmv2/modeling_layoutlmv2.py [hunk 1]
```diff
@@ -0,0 +1,1023 @@
...
+
+            >>> outputs = model(input_ids=input_ids, bbox=bbox, attention_mask=attention_mask, token_type_ids=token_type_ids,
+            ...                 labels=sequence_label)
+
+            >>> loss = outputs.loss
+            >>> logits = outputs.logits
+        """
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        outputs = self.layoutlm(
```
Comment (dcyoung) [inline src/transformers/models/layoutlmv2/modeling_layoutlmv2.py:986]: It appears that the image is not propagated to this forward call? Is there a reason the image is not used in sequence classification?

File: setup.py [hunk 2]
```diff
@@ -323,6 +324,7 @@ def run(self):
     deps["sacremoses"],  # for XLM
     deps["tokenizers"],
     deps["tqdm"],  # progress bars in model download and training scripts
+    deps["detectron2"],  # detectron2 for image embedding
```
Comment (raguiar2) [inline setup.py:327]: use 
`deps["detectron2 @ git+https://github.com/facebookresearch/detectron2.git"], # detectron2 for image embedding`

as documented in https://stackoverflow.com/questions/60414753/how-to-install-githttps-from-setup-py-using-install-requires

File: setup.py [hunk 3]
```diff
@@ -149,6 +149,7 @@
     "unidic>=1.0.2",
     "unidic_lite>=1.0.7",
     "uvicorn",
+    "detectron2",
```
Comment (raguiar2) [inline setup.py:152]: use `detectron2 @ git+https://github.com/facebookresearch/detectron2.git`

File: setup.py (hunk 2 above)
Comment (monuminu) [reply setup.py:327]: still giving error saying that torch is not available . any help please

Comment (raguiar2) [reply setup.py:327]: You need to change this at all points where detectron2 is used. Did you run the tests locally? Also, it's `detectron2 @ git+https://github.com/facebookresearch/detectron2.git` (with a space). I will tag where you have to replace it

File: setup.py [hunk 4]
```diff
@@ -159,6 +159,7 @@
 #
 # some of the values are versioned whereas others aren't.
 deps = {b: a for a, b in (re.findall(r"^(([^!=<>]+)(?:[!=<>].*)?$)", x)[0] for x in _deps)}
+deps["detectron2"] = "detectron2@git+https://github.com/facebookresearch/detectron2.git"
```
Comment (raguiar2) [inline setup.py:162]: this isn't right - the string ``detectron2 @ git+https://github.com/facebookresearch/detectron2.git` should be above right below "uvicorn"

File: src/transformers/dependency_versions_table.py [hunk 5]
```diff
@@ -66,4 +66,5 @@
     "unidic": "unidic>=1.0.2",
     "unidic_lite": "unidic_lite>=1.0.7",
     "uvicorn": "uvicorn",
+    "detectron2" : "detectron2",
```
Comment (raguiar2) [inline src/transformers/dependency_versions_table.py:69]: This has to be `"detectron2 @ git+https://github.com/facebookresearch/detectron2.git": "git+https://github.com/facebookresearch/detectron2.git@v0.4#egg=detectron2",` not `"detectron2" : "detectron2",`

Comment (monuminu) [reply src/transformers/dependency_versions_table.py:69]: done still erroring out . any thought ?

Comment (raguiar2) [reply src/transformers/dependency_versions_table.py:69]: This isnt changed? my comment said for this to be  `"detectron2 @ git+https://github.com/facebookresearch/detectron2.git": "git+https://github.com/facebookresearch/detectron2.git@v0.4#egg=detectron2",`

Comment (raguiar2) [reply src/transformers/dependency_versions_table.py]: Oh, it's still failing? Weird, mine builds locally totally fine

**Pull request closed without merging**